### ISSUE TITLE
Loss function and large angles

### ISSUE THREAD

**Issue by paulremondeau** [pidahbus/deep-image-orientation-angle-detection#4]
Hello, and thank you very much for your repo, it is really useful to me. I have one question though about the loss function you used for training.

According to your paper and your code, the loss function on which the model is optimised is $min(|p-t|, 360 - |p-t|)$. This seems great to me **if** $p$ and $t$ are in $[0,360]$, or $[-180, 180]$ in our case.

I have however encountered a negative loss function during my training, which lead me to the following question : **can the model be actually tempted to predict really large angles** ? For example if my true angle is 40 and it predicts -600, my angular loss would be -200.

What I saw during training :
![image](https://user-images.githubusercontent.com/55699126/220108300-7ce63604-6042-406b-8dab-33ddae1adf02.png)

My validation csv file, wth angles between -180 and 180 : 
![image](https://user-images.githubusercontent.com/55699126/220108990-05235a10-8243-41f1-bd2a-05275e1248dd.png)

**Comment by pidahbus** [comment]
Hi, convert your model angles between 0 to 360 and then train the model.

**Comment by pidahbus** [comment]
Hi, I have released version 2 of our model with VIT Implementation and dynamic training data generation. Feel free to look at it and ask if you have any queries.
